Source/wayland/WaylandServer.m

```diff
@@ -0,0 +1,1482 @@
+	       window->width, window->height);
+
+    return NSMakeRect(window->pos_x, window->output->height - window->pos_y,
+		      window->width, window->height);
+}
+
+- (void) setwindowlevel: (int) level : (int) win
+{
+    NSDebugLog(@"setwindowlevel: level=%d win=%d", level, win);
+}
+
+- (int) windowlevel: (int) win
+{
+    NSDebugLog(@"windowlevel: %d", win);
+    return 0;
+}
+
+/** Backends can override this method to return an array of window numbers
+    ordered front to back.  The front most window being the first object
+    in the array.
+    The default implementation returns the visible windows in an
+    unspecified order.
+ */
+- (NSArray *) windowlist
+{
+  NSMutableArray *list = [NSMutableArray arrayWithArray:[NSApp windows]];
+  int c = [list count];
+
+  while (c-- > 0)
+    {
+       if (![[list objectAtIndex:c] isVisible])
+         {
+	   [list removeObjectAtIndex:c];
+         }
+    }
+  return [list valueForKey:@"windowNumber"];
+}
+
+- (int) windowdepth: (int) win
+{
+    NSDebugLog(@"windowdepth");
+    return 0;
+}
+
+- (void) setmaxsize: (NSSize)size : (int) win
+{
+    NSDebugLog(@"setmaxsize");
+}
+
+- (void) setminsize: (NSSize)size : (int) win
+{
+    NSDebugLog(@"setminsize");
+}
+
+- (void) setresizeincrements: (NSSize)size : (int) win
+{
+    NSDebugLog(@"setresizeincrements");
+}
+
+- (void) flushwindowrect: (NSRect)rect : (int) win
+{
+    NSDebugLog(@"flushwindowrect: %d %fx%f", win, NSWidth(rect), NSHeight(rect));
+    struct window *window = get_window_with_id(wlconfig, win);
+
+    [[GSCurrentContext() class] handleExposeRect: rect forDriver: window->wcs];
+}
+
+- (void) styleoffsets: (float*) l : (float*) r : (float*) t : (float*) b
+		     : (unsigned int) style
+{
+    NSDebugLog(@"styleoffsets");
+    /* XXX - Assume we don't decorate */
+    *l = *r = *t = *b = 0.0;
+}
+
+- (void) docedited: (int) edited : (int) win
+{
+    NSDebugLog(@"docedited");
+}
+
+- (void) setinputstate: (int)state : (int)win
+{
+    NSDebugLog(@"setinputstate");
+}
+
+- (void) setinputfocus: (int) win
+{
+    NSDebugLog(@"setinputfocus");
+}
+
+- (void) setalpha: (float)alpha : (int) win
+{
+    NSDebugLog(@"setalpha");
+}
+
+- (void) setShadow: (BOOL)hasShadow : (int)win
+{
+    NSDebugLog(@"setshadow");
+}
+
+- (NSPoint) mouselocation
+{
+    NSDebugLog(@"mouselocation");
+    return NSZeroPoint;
+}
+
+- (NSPoint) mouseLocationOnScreen: (int)aScreen window: (int *)win
+{
+    NSDebugLog(@"mouseLocationOnScreen: %d %fx%f", win,
+	       wlconfig->pointer.x, wlconfig->pointer.y);
+    struct window *window = wlconfig->pointer.focus;
+    struct output *output;
+    float x;
+    float y;
+
+    /*if (wlconfig->pointer.serial) {
+	NSDebugLog(@"captured");
+	x = wlconfig->pointer.captured_x;
+	y = wlconfig->pointer.captured_y;
+	} else*/ {
+	NSDebugLog(@"NOT captured");
+	x = wlconfig->pointer.x;
+	y = wlconfig->pointer.y;
+
+	if (window) {
+	    x += window->pos_x;
+	    y += window->pos_y;
+	    win = &window->window_id;
```

Comment:
Check for win being NULL is missing.

Reply:
Interestingly, it was not even writing into `*win`...